app/controllers/BaseController.java

```diff
@@ -64,17 +68,25 @@ public void setBridgeConfig(BridgeConfig bridgeConfig) {
     }
     
     /**
+     * Returns a session. Will not throw exception if user is not authorized or has not consented to research.
+     * @return session if it exists, or null otherwise.
+     */
+    protected UserSession getSessionIfItExists() {
+        String sessionToken = getSessionToken();
+        return authenticationService.getSession(sessionToken);
```

Comment:
What happens if you call getSession with a null or empty session token? Should have have a check here? Or possibly in authService?